$NetBSD: patch-CMakeLists.txt,v 1.1 2025/10/02 11:49:44 nia Exp $

More reliable mechanism to find libuv.

Don't insist on ncurses.

--- CMakeLists.txt.orig	2017-11-10 12:28:57.000000000 +0000
+++ CMakeLists.txt
@@ -544,7 +544,8 @@ int main(void) { return 0; }
   if(CMAKE_USE_LIBUV)
     if(CMAKE_USE_SYSTEM_LIBUV)
       if(NOT CMAKE_VERSION VERSION_LESS 3.0)
-        find_package(LibUV 1.0.0)
+        include(FindPkgConfig)
+        pkg_check_modules(LIBUV libuv)
       else()
         message(FATAL_ERROR "CMAKE_USE_SYSTEM_LIBUV requires CMake >= 3.0")
       endif()
@@ -552,7 +553,7 @@ int main(void) { return 0; }
         message(FATAL_ERROR
           "CMAKE_USE_SYSTEM_LIBUV is ON but a libuv is not found!")
       endif()
-      set(CMAKE_LIBUV_LIBRARIES LibUV::LibUV)
+      set(CMAKE_LIBUV_LIBRARIES ${LIBUV_LIBRARIES})
     else()
       set(CMAKE_LIBUV_LIBRARIES cmlibuv)
       add_subdirectory(Utilities/cmlibuv)
@@ -579,7 +580,6 @@ int main(void) { return 0; }
   if (UNIX)
     # there is a bug in the Syllable libraries which makes linking ccmake fail, Alex
     if(NOT CMAKE_SYSTEM_NAME MATCHES syllable)
-      set(CURSES_NEED_NCURSES TRUE)
       find_package(Curses QUIET)
       if (CURSES_LIBRARY)
         option(BUILD_CursesDialog "Build the CMake Curses Dialog ccmake" ON)
